src/Context/GtmContext.php

```diff
@@ -75,17 +76,69 @@ public function getDataLayerSettingShouldMatch($key, $regex) {
    * @return mixed
    * @throws \Exception
    */
-  protected function getDataLayerValue($key) {
-    $json_arr = $this->getDataLayerJson();
+    protected function getDataLayerValue($key)
+    {
+        $json_arr = $this->getDataLayerJson();
+        // Loop through the array and return the data layer value.
+        foreach ($json_arr as $json_item) {
+            // Check if the key contains dot.
+            if (strpos($key, '.', 0) !== false) {
+                // Get value using dot notation.
+                $value = $this->getDotValue($json_item, $key);
+                if (!is_null($value)) {
+                    return $value;
+                }
+            } elseif (isset($json_item[$key])) {
+                return $json_item[$key];
+            }
+        }
+        throw new \Exception($key . ' not found.');
+    }
 
-    // Loop through the array and return the data layer value
-    foreach ($json_arr as $json_item) {
-      if (isset($json_item[$key])) {
-        return $json_item[$key];
-      }
+    /**
+     * Convert arrays :arg1 to dot notation :arg2
+     */
+    private function getDotValue($value, $key)
+    {
+        $dot = dot($value);
+        // Check if the key contains [.
+        // When the key contains [0] it means that we are trying to get the value from specific index in an array.
+        // i.e. foo.bar[0].foo.
+        $pos = strpos($key, '[',0);
+        if ($pos) {
+            // Trim before [
+            $keyBeforeArray = substr($key, 0, $pos);
+            $values = $dot->get($keyBeforeArray);
+            if (!empty($values)) {
+                // Filter the number from string.
+                $int_var = $this->getNumberInsideKey($key);
```

Comment:
Please do not mix Camel Case and Snake Case, all variables should follow same pattern. I would recommend you to rename this variable to $index.